src/main/java/org/skriptlang/skript/lang/script/Script.java

```diff
@@ -65,6 +67,7 @@ public Config getConfig() {
 	/**
 	 * @return An unmodifiable list of all Structures within this Script.
 	 */
+	@Unmodifiable
 	public List<Structure> getStructures() {
 		return Collections.unmodifiableList(structures);
```

Comment:
Rather than doing this every time. Use ImmutableSet.copyOf(...) in the constructor. See https://github.com/SkriptLang/Skript/pull/5317#discussion_r1101973377

Reply:
Due to some restrictions in package organization, we can't make the copy in the constructor as we need the list to be modifiable for initial setup. This setup is not ideal, but it results in the "best" protection against unwanted modifications with what we have. See https://github.com/SkriptLang/Skript/pull/5321#discussion_r1083141955

However, it might be a good idea to have this return a copy/snapshot view instead of the live view.